community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/ast/convert/commands/StatementConverters.scala

```diff
@@ -231,13 +231,15 @@ object StatementConverters {
   }
 
   implicit class RonjaHintConverter(val item: ast.UsingHint) extends AnyVal {
-    def asCommandStartHint(notifications: InternalNotificationLogger) = item match {
+    def asCommandStartHint(notifications: InternalNotificationLogger, plannerName: String) = item match {
       case ast.UsingIndexHint(identifier, label, property) =>
         Some(commands.SchemaIndex(identifier.name, label.name, property.name, commands.AnyIndex, None))
       case ast.UsingScanHint(identifier, label) =>
         Some(commands.NodeByLabel(identifier.name, label.name))
       case ast.UsingJoinHint(identifiers) =>
-        notifications.log(JoinHintUnsupportedNotification(identifiers.map(_.name).toSeq))
+        if (PlannerName(plannerName) == RulePlannerName) {
```

Comment:
This is only generating commands for the rule planner. I would guess the reason it is called also for cost planner is for the sake of simplicity: we sometimes need to fallback to rule planner, and since this is called in an early stage, the alternative to go back and produce this only when we realise that we actually need the fallback may be much more difficult.